atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -86,11 +89,31 @@ private static ThreadFactory createThreadFactory(String safeLoggablePurpose) {
     public ListenableFuture<R> apply(T argument) {
         Preconditions.checkState(!closed, "Autobatcher is already shut down");
         DisruptorFuture<R> result = new DisruptorFuture<R>(safeLoggablePurpose);
-        buffer.publishEvent((refresh, sequence) -> {
+        buffer.publishEvent(createTranslator(argument, result));
+        return result;
+    }
+
+    public List<ListenableFuture<R>> applyBatch(List<T> arguments) {
```

Comment:
Would it be a lot of overhead for removing the non-batched method here?

Reply:
I think so - it is used in quite a few places.

We could instead just have `apply` call `applyBatch` instead.